File: go/database/table/border/border.go
package border

type Border struct {
	CornerUpLeft      string
	CornerUpRight     string
	CornerBottomLeft  string
	CornerBottomRight string
	DivisionUp        string
	DivisionBottom    string
	Horizontal        string
	Vertical          string
	Intersection      string
	VerticalLeft      string
	VerticalRight     string
}

func GetBorder(opt BorderOption) Border {
	return borders[opt]
}

type BorderOption int

const (
	Default BorderOption = iota + 1
	Simple
	Rounded
	Double
	SimpleDouble
)

var borders map[BorderOption]Border

func init() {
	borders = map[BorderOption]Border{
		1: {
			CornerUpLeft:      "┏",
			CornerUpRight:     "┓",
			CornerBottomLeft:  "┗",
			CornerBottomRight: "┛",
			DivisionUp:        "┳",
			DivisionBottom:    "┻",
			Horizontal:        "━",
			Vertical:          "┃",
			Intersection:      "╋",
			VerticalLeft:      "┣",
			VerticalRight:     "┫",
		},
		2: {
			CornerUpLeft:      "┌",
			CornerUpRight:     "┐",
			CornerBottomLeft:  "└",
			CornerBottomRight: "┘",
			DivisionUp:        "┬",
			DivisionBottom:    "┴",
			Horizontal:        "─",
			Vertical:          "│",
			Intersection:      "┼",
			VerticalLeft:      "├",
			VerticalRight:     "┤",
		},
		3: {
			CornerUpLeft:      "╭",
			CornerUpRight:     "╮",
			CornerBottomLeft:  "╰",
			CornerBottomRight: "╯",
			DivisionUp:        "┬",
			DivisionBottom:    "┴",
			Horizontal:        "─",
			Vertical:          "│",
			Intersection:      "┼",
			VerticalLeft:      "├",
			VerticalRight:     "┤",
		},
		4: {
			CornerUpLeft:      "╔",
			CornerUpRight:     "╗",
			CornerBottomLeft:  "╚",
			CornerBottomRight: "╝",
			DivisionUp:        "╦",
			DivisionBottom:    "╩",
			Horizontal:        "═",
			Vertical:          "║",
			Intersection:      "╬",
			VerticalLeft:      "╠",
			VerticalRight:     "╣",
		},
		5: {
			CornerUpLeft:      "╒",
			CornerUpRight:     "╕",
			CornerBottomLeft:  "╘",
			CornerBottomRight: "╛",
			DivisionUp:        "╤",
			DivisionBottom:    "╧",
			Horizontal:        "═",
			Vertical:          "│",
			Intersection:      "╪",
			VerticalLeft:      "╞",
			VerticalRight:     "╡",
		},
	}
}

File: java/src/main/java/tabula/database/table/Border.java
package tabula.database.table;

import java.util.HashMap;
import java.util.Map;

public record Border(String cornerUpLeft, String cornerUpRight, String cornerBottomLeft, String cornerBottomRight,
        String divisionUp, String divisionBottom, String horizontal,
        String vertical, String intersection, String verticalLeft, String verticalRight) {

    public enum BorderOption {
        DEFAULT(1),
        SIMPLE(2),
        ROUNDED(3),
        DOUBLE(4),
        SIMPLE_DOUBLE(5);

        private final int value;

        BorderOption(int value) {
            this.value = value;
        }

        public int getValue() {
            return value;
        }

        public static BorderOption get(int value) {
            for (BorderOption bo : BorderOption.values()) {
                if (bo.value == value) {
                    return bo;
                }
            }
            return null;
        }
    }

    private static final Map<BorderOption, Border> borders = new HashMap<>();

    static {
        borders.put(BorderOption.DEFAULT, new Border("┏", "┓", "┗", "┛", "┳", "┻", "━", "┃", "╋", "┣", "┫"));
        borders.put(BorderOption.SIMPLE, new Border("┌", "┐", "└", "┘", "┬", "┴", "─", "│", "┼", "├", "┤"));
        borders.put(BorderOption.ROUNDED, new Border("╭", "╮", "╰", "╯", "┬", "┴", "─", "│", "┼", "├", "┤"));
        borders.put(BorderOption.DOUBLE, new Border("╔", "╗", "╚", "╝", "╦", "╩", "═", "║", "╬", "╠", "╣"));
        borders.put(BorderOption.SIMPLE_DOUBLE, new Border("╒", "╕", "╘", "╛", "╤", "╧", "═", "│", "╪", "╞", "╡"));
    }

    public static Border getBorder(int option) {
        return borders.get(BorderOption.get(option));
    }
}
